go/oasis-node/cmd/common/metrics/cpu.go

```diff
@@ -0,0 +1,79 @@
+// Package metrics implements a prometheus metrics service.
+package metrics
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+)
+
+const (
+	MetricCPUUTimeSeconds = "oasis_worker_cpu_utime_seconds"
+	MetricCPUSTimeSeconds = "oasis_worker_cpu_stime_seconds"
+
+	// getconf CLK_TCK
+	ClockTicks = 100
+)
+
+var (
+	utimeGauge prometheus.Gauge
+	stimeGauge prometheus.Gauge
+
+	cpuServiceOnce sync.Once
+)
+
+type cpuCollector struct {
+	// TODO: Should we monitor memory of children PIDs as well?
+	pid int
+}
+
+func (c *cpuCollector) Update() error {
+	// Obtain process CPU info.
+	proc, err := procfs.NewProc(c.pid)
+	if err != nil {
+		return fmt.Errorf("CPU metric: failed to obtain proc object for PID %d: %v", c.pid, err)
+	}
+	procStat, err := proc.Stat()
+	if err != nil {
+		return fmt.Errorf("CPU metric: failed to obtain procStat object %d: %v", c.pid, err)
+	}
+
+	utimeGauge.Set(float64(procStat.UTime) / float64(ClockTicks))
+	stimeGauge.Set(float64(procStat.STime) / float64(ClockTicks))
+
+	return nil
+}
+
+// NewCPUService constructs a new memory usage service.
```

Comment:
```suggestion
// NewCPUCollector constructs a new CPU usage collector.
```

(similar Service->Collector for other collectors)